framework/include/base/NonlinearEigenSystem.h

```diff
@@ -91,8 +81,19 @@ class NonlinearEigenSystem
   virtual TransientEigenSystem & sys() { return _transient_sys; }
 protected:
   TransientEigenSystem & _transient_sys;
+#endif /* LIBMESH_HAVE_SLEPC */
+protected:
   std::vector<std::pair<Real, Real> > _eigen_values;
   unsigned int _n_eigen_pairs_required;
-#endif /* LIBMESH_HAVE_SLEPC */
+public:
+  // For eigenvalue problems (including standard and generalized), inhomogeneous (Dirichlet or Neumann)
+  // boundary conditions are  not allowed.
+  void checkIntegrity();
+  // return the Nth converged eigenvlue <real, imag>
```

Comment:
Sorry for confusing you! What I want to say is that the pair <Real , Real> represents an eigenvalue. The eigenvalue could be a complex, and if this is the case, the second member of the pair is the image part of the complex value, otherwise, the second part is zero. Comments `imag` means the image part of the complex.